File: LeetcodeQuestions/leetcode_challenges/February_challange/arithmetic_slices.rb
def number_of_arithmetic_slices(a)
    if a.length < 3
        return 0
    end
    n = a.length
    dp = Array.new(n).fill(0)
    num_slices = 0
    
    (2...n).each do |i|
        if a[i] - a[i - 1] == a[i - 1] - a[i - 2]
            dp[i] = dp[i - 1] + 1
        end
        num_slices += dp[i]
    end
    return num_slices
end

File: LeetcodeQuestions/leetcode_challenges/February_challange/arithmeticSlices.js
function numberOfArithmeticSlices(nums) {
    if (nums.length < 3) return 0;
    let n = nums.length;
    let dp = new Array(n).fill(0);
    let numSlices = 0;
    for (let i = 2; i < n; i++) {
        if (nums[i] - nums[i - 1] === nums[i - 1] - nums[i - 2]) {
            dp[i] = dp[i - 1] + 1;
        }
        numSlices += dp[i]
    }
    return numSlices;
}

console.log(numberOfArithmeticSlices([1, 2, 3, 4]))